Client/Frontend/Browser/URIFixup.swift

```diff
@@ -34,4 +39,23 @@ class URIFixup {
 
         return nil
     }
-}
\ No newline at end of file
+
+    static func punycodedStr(string : String) -> String {
+        var host = string
+
+        if let p = host.rangeOfString("://") {
+            host = host.substringFromIndex(p.endIndex)
+        }
+
+        if let t = host.rangeOfString("/") {
+            host = host.substringToIndex(t.startIndex)
+        }
+
+        if let s = host.rangeOfString(":") {
+            host = host.substringToIndex(s.startIndex)
+        }
+
+        let urlStr = string.stringByReplacingOccurrencesOfString(host, withString: host.utf8HostToAscii())
```

Comment:
I totally agree with you.
And I have changed this code according to your suggestion.